docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md

```diff
@@ -0,0 +1,179 @@
+# Using Foundry with the Avalanche C-Chain
+
+## Introduction
+
+This article shows how to deploy and interact with smart contracts using foundry on the [Fuji C-Chain](../../quickstart/fuji-workflow.md).
+
+[Foundry](https://github.com/foundry-rs/foundry) is a smart contract development toolchain written in Rust.
+
+Foundry manages your dependencies, compiles your project, runs tests, deploys, and lets you interact with the chain from the command-line.
+
+## Prerequisites
+
+- You have installed [Foundry](https://github.com/foundry-rs/foundry#installation); This installation includes the `forge` and `cast` binaries used in this walk-through
+- You are familiar with [Avalanche Smart Contract Quickstart](https://github.com/ava-labs/avalanche-smart-contract-quickstart)
+- If you plan on running locally, ensure that you have installed and are familiar with [Avalanche Network Runner](../../quickstart/network-runner)
+## Getting Started
+
+This section will walk you through creating an ERC721 with Foundry and Avalanche Smart Contract Quickstart.
+Clone the Avalanche Smart Contract Quickstart repo and install its dependencies by running:
+
+```zsh
+git clone https://github.com/ava-labs/avalanche-smart-contract-quickstart.git
+cd avalanche-smart-contract-quickstart
+yarn
+```
+
+In order to deploy contracts, you need to have some AVAX. You can get testnet AVAX from the [Avalanche Faucet](https://faucet.avax.network), which is an easy way to get to play around with Avalanche. After getting comfortable with your code, you can run it on Mainnet after making the necessary changes to your workflow.
+
+## Implement a Game Item NFT
+
+We will use our example ERC721 smart contract, [`NFT.sol`](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/3fbba0ac28f6420e9be5d2635d5f23693f80127a/contracts/NFT.sol) found in `./contracts` of our project.
+
+```ts
```

Comment:
Can we have this block of text be of type `solidity` ? I'm not sure if that works. If not then `ts` is good.

Reply:
solidity is not yet supported :(